src/signature.c

```diff
@@ -46,115 +46,301 @@
 
 #warning "TODO pick signing scheme"
 #if defined(SWUPD_LINUX_ROOTFS)
-#define IMPL IMPL_FAKE
+#define IMPL IMPL_REAL
```

Comment:
Why would this be anything other than IMPL_REAL (or an option at all?) Isn't checking the signatures controlled by the command line option? An incorrect build config could easily cause these checks to be bypassed, while telling the calling script that everything is good.